_test/test_algorithms/test_duplicated_id.m

```diff
@@ -0,0 +1,100 @@
+classdef test_duplicated_id < TestCase
+    
+    properties
+    end
+    
+    methods
+        
+        function obj = test_duplicated_id(varargin)
+            if nargin > 0
+                name = varargin{1};
+            else
+                name= mfilename('class');
+            end
+            obj = obj@TestCase(name);
+        end
+        function test_update_duplicates_no_empty(~)
+            %
+            runfiles = {rundatah(),rundatah(),...
+                rundatah(),rundatah(),rundatah(),...
+                rundatah(),rundatah(),rundatah(),rundatah()};
+            % create 1,1,2,2,2,3,3,3,3 runid sequence
+            runids = [1,1,2,2,2,3,3,3,3];
+            for i=1:numel(runfiles)
+                runfiles{i}.run_id = runids(i);
+            end
+            rf_proc = update_duplicated_rf_id(runfiles);
+            % Got the sequence from runid
+            ids = cellfun(@(x)(x.run_id),rf_proc,'UniformOutput',true);
+            rd_unique = [1,4,2,5,6,3,7,8,9];
+            assertEqual(ids,rd_unique);
+        end
+        
+        
+        function test_update_duplicates(~)
+            %
+            runfiles = {rundatah(),[],rundatah(),rundatah(),[],...
+                rundatah(),rundatah(),rundatah()};
+            ic = 1;
+            count = 1;
+            % create 1,'',2,2,'',3,3,3 runid sequence
+            for i=1:numel(runfiles)
+                if isempty(runfiles{i})
+                    continue;
+                end
+                if ic>count
+                    count = count + 1;
+                    ic = 1;
+                end
+                runfiles{i}.run_id = count;
+                ic = ic+1;
+            end
+            rf_proc = update_duplicated_rf_id(runfiles);
+            % Got 1,0,2,6,0,3,7,8 sequence
+            function id = get_id(run)
+                if isempty(run)
+                    id = 0;
+                else
+                    id = run.run_id;
+                end
+            end
+            ids = cellfun(@get_id,rf_proc,'UniformOutput',true);
+            assertEqual(ids,[1,0,2,6,0,3,7,8]);
+        end
+        
+        function test_do_nothing_inverted(~)
```

Comment:
It's not clear from the names why this these do_nothing tests are expected to do nothing

Reply:
I wrote comment, explaining why this behavior is expected

Reply:
you can use the test name to make that clear too, e.g. `test_does_nothing_if_all_data_unique`, `test_does_nothing_if_data_unique_and_inverse_order`